impress-org/givewp
# Feature: add list table for campaigns

Resolves [GIVE-1151]

## Description



This PR implements the campaigns list table and a few API routes to implement CRUD operations.

The new UI allows users to view, create, update, and delete campaigns. 

We still need to work on the design, form inputs, and the columns that we want to display in the list table - it should be done in subsequent PRs.

## Affects



The campaigns admin page.

## Visuals



![image](https://github.com/user-attachments/assets/d98f5ee6-c024-420f-8159-2f44f54c994c)

![image](https://github.com/user-attachments/assets/cba61ded-0d7c-44f6-8a2f-e800d238c781)

![image](https://github.com/user-attachments/assets/0652510f-b32f-4469-91db-bc5a3ad8a0ac)

![image](https://github.com/user-attachments/assets/6fbd8b2a-9c21-4e0c-81ab-eae22bdf380e)

## Testing Instructions

1. Go to `Give > Campaigns` page;
2. Create a few campaigns;
3. Use the Search box to filter a specific campaign;
4. Use the Delete action to remove a specific campaign;
5. Make sure everything worked as expected.



## Pre-review Checklist






[GIVE-1151]: https://stellarwp.atlassian.net/browse/GIVE-1151?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
